packages/protocol/contracts/L1/provers/Guardians.sol

```diff
@@ -48,34 +49,41 @@ abstract contract Guardians is EssentialContract {
         onlyOwner
         nonReentrant
     {
+        // We need at least MIN_NUM_GUARDIANS and at most 255 guardians (so the approval bits fit in
+        // a uint256)
         if (_newGuardians.length < MIN_NUM_GUARDIANS || _newGuardians.length > type(uint8).max) {
             revert INVALID_GUARDIAN_SET();
         }
-        if (_minGuardians < _newGuardians.length >> 1 || _minGuardians > _newGuardians.length) {
+        // Minimum number of guardians to approve is at least equal or greater than half the
+        // guardians (rounded up) and less or equal than the total number of guardians
+        if (_minGuardians < (_newGuardians.length + 1) >> 1 || _minGuardians > _newGuardians.length)
+        {
             revert INVALID_MIN_GUARDIANS();
         }
 
-        // Delete current guardians data
```

Comment:
Also just some simplifications here below. This code seems written like it tries to optimize gas as much as possible, but this code will almost never be called, so simplicity should be preferred.